### PR TITLE
Extrapolation

### PR DESCRIPTION
**Pull request by HugoGranstrom**
I've written a short section on the new extrapolation features of numericalnim's 1D interpolators. And I've also fixed a few links and updated to the latest nimib/ook. Let's hope it runs, and it should be faster now with parallel builds.

### CODE REVIEW

File: book/numerical_methods/interpolation.nim [hunk 1]
```diff
@@ -92,6 +92,55 @@ We can now evaluate it on a denser set of points and compare it to the original
...
+
+  nbCode:
+    echo "Edge: ", interp.eval(-1.0, Edge) # will use the value at x=0
+    echo "Constant: ", interp.eval(-1.0, Constant, NaN) # will return NaN
+    echo "Linear: ", interp.eval(-1.0, ExtrapolateKind.Linear)
+    echo "Native: ", interp.eval(-1.0) # will use Native by default
+
+  nbText: hlMd"""
+As you can see, the choice of extrapolation method affects the values considerably.
+Keep in mind though that `-1.0` is quite a big extrapolation and the further away we go, the worse the approximation get.
```
Comment (Vindaar) [inline book/numerical_methods/interpolation.nim:115]:
```suggestion
Keep in mind though that `-1.0` is quite a big extrapolation and the further away we go, the worse the approximation gets.
```
(or "approximations get")

File: book/numerical_methods/interpolation.nim [hunk 2]
```diff
@@ -92,6 +92,55 @@ We can now evaluate it on a denser set of points and compare it to the original
...
+
+Here is a visual example of how the different methods behave. I have removed the two othermost points on each side from the example before (ignore the point at `(0, 0)`, it's a bug):
```
Comment (Vindaar) [inline book/numerical_methods/interpolation.nim:117]:
```suggestion
Here is a visual example of how the different methods behave. I have removed the two outermost points on each side from the example before (ignore the point at `(0, 0)`, it's a bug):
```
I assume?

**Comment by Vindaar** [comment]
Looking good, thanks!

**Comment by HugoGranstrom** [comment]
Thank you! :D I really should install something like LanguageTool in VsCodium so you wouldn't have to correct my language in every review :stuck_out_tongue_winking_eye: 

I'll leave this open and merge it tonight unless anyone else has any feedback.

**Comment by HugoGranstrom** [comment]
Ah, I need that sweet Approval to merge this :sweat_smile:

**Comment by HugoGranstrom** [comment]
Thanks Clonkk 😄

**Pull request merged**